qhub/utils.py

```diff
@@ -345,6 +345,22 @@ def keycloak_provider_context(keycloak_credentials: Dict[str, str]):
         yield
 
 
+@contextlib.contextmanager
+def terraform_state_context(provider: str, terraform_state: str):
+    credentials = {}
+
+    if provider == "do" and terraform_state == "remote":
+        credentials.update(
+            {
+                "AWS_ACCESS_KEY_ID": os.eviron["SPACES_ACCESS_KEY_ID"],
```

Comment:
So, basically, we use `AWS` when we need to provide a remote backend for DO, is that right? This means that if the user wants to use `remote` for a DO deployment it also would need to have an AWS account and those variables in hand, right?

Reply:
Digital Ocean spaces is a compliant S3 object storage. The reason we have to do this is due to a limitation/assumption in the terraform remote state. We are not actually using anything related to AWS here but making it look to terraform remote state

Reply:
> due to a limitation/assumption in the terraform remote state

Reply:
https://www.terraform.io/language/settings/backends/s3